frontend/src/services/graphQL.ts

```diff
@@ -0,0 +1,115 @@
+import axios from 'axios'
+import { r3 } from '../services/remote.it'
+import { parseType } from '../services/serviceTypes'
+import { renameServices } from '../helpers/nameHelper'
+import { GRAPHQL_API_URL } from '../constants'
+import { updateConnections } from '../helpers/connectionHelper'
+
+export async function graphQLFetch({ size, from, state = '', name = '', ids = [] }: gqlOptions) {
+  const connections = `connections: devices(id:${JSON.stringify(ids)})`
+  const devices = `devices(size:${size}, from:${from}, name:"${name}", state: "${state}")`
+  const select = `
+  {
+    total
+    items {
+      id
+      name
+      state
+      created
+      hardwareId
+      lastReported
+      endpoint {
+        externalAddress
+        internalAddress
+        serverAddress
+        availability
+        instability
+        geo {
+          country
+          state
+          isp
+        }
+      }
+      owner {
+        id
+        email
+      }
+      services {
+        name
+        id
+        port
+        type
```

Comment:
application (which is the number part like 35 for bulk)

                "type": "00:1C:00:00:00:04:00:06:04:30:00:16:00:01:00:00",
                "application": 28,
                "title": "SSH"